Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/EwsExtensionEXOPowershellV3.ps1

```diff
@@ -0,0 +1,1572 @@
+    )
+    {
+        $cmd_params = @{ }
+        if ($identity)
+        {
+            $cmd_params.Identity = $identity
+        }
+        if ($property_sets)
+        {
+            $cmd_params.PropertySets = $property_sets
+        }
+        $this.CreateSession()
+        if ($limit -gt 0)
+        {
+            $results = Get-EXOMailbox @cmd_params -ResultSize $limit
+        }
+        else
+        {
+            $results = Get-EXOMailbox @cmd_params -ResultSize Unlimited
+        }
+        $this.DisconnectSession()
+        return $results
+        <#
+        .DESCRIPTION
+        Use the Get-EXOMailbox cmdlet to view mailbox objects and attributes,
+        populate property pages, or supply mailbox information to other tasks.
+        This cmdlet is available only in the Exchange Online PowerShell V2 module
+
+        .PARAMETER identity
+        The Identity parameter specifies the mailbox you want to view.
+        For the best performance, we recommend using the following values:
+            * User ID or user principal name (UPN)
+            * GUID
+
+        Otherwise, you can use any value that uniquely identifies the mailbox. For example:
+            * Name
+            * Alias
+            * Distinguished name (DN)
+            * Domain\Username
+            * Email address
+            * LegacyExchangeDN
+            * SamAccountName
+
+        .EXAMPLE
+        GetEXOMailbox("1254y894-feae-9yn7-a3e1-f2483a154tft")
+
+        .OUTPUTS
+        PSObject - Raw response
+
+        .LINK
+        https://docs.microsoft.com/en-us/powershell/module/exchange/get-exomailbox?view=exchange-ps
+        #>
+    }
+
+    [PSObject]
+    GetEXOMailboxPermission(
+            [string]$identity
+    )
+    {
+        # Import and Execute command
+        $cmd_params = @{ }
+        if ($identity)
+        {
+            $cmd_params.Identity = $identity
+        }
+        $this.CreateSession()
+        $results = Get-EXOMailboxPermission @cmd_params
+        $this.DisconnectSession()
+        return $results
+
+        <#
+        .DESCRIPTION
+        View information about SendAs permissions that are configured for users.
+        This command is available only in the Exchange Online PowerShell V2 module.
+
+        .PARAMETER identity
+        The Identity parameter the user that you want to view. You can use any value that uniquely identifies the user
+
+        .EXAMPLE
+        GetEXOMailboxPermission("1254y894-feae-9yn7-a3e1-f2483a154tft")
+
+        .OUTPUTS
+        PSObject - Raw response
+
+        .LINK
+        https://docs.microsoft.com/en-us/powershell/module/exchange/get-exomailboxpermission?view=exchange-ps
+        #>
+    }
+    [PSObject]
+    GetEXORecipientPermission(
+            [string]$identity,
+            [int]$limit
+    )
+    {
+        $cmd_params = @{ }
+        if ($identity)
+        {
+            $cmd_params.Identity = $identity
+        }
+        $this.CreateSession()
+        if ($limit -gt 0)
```

Comment:
perhaps tighten it to call the command just once with cmd_params:
`if ($identity)
{
            $cmd_params.Identity = $identity
}
if ($limit -gt 0)
{
       $cmd_params.ResultSize = $limit
}
else 
{
       $cmd_params.ResultSize = Unliimited
}
$results = Get-EXORecipientPermission @cmd_params
`